lib/puppet_x/templates/invoke_ps_command.erb

```diff
@@ -1,15 +1,255 @@
 $hostSource = @"
 using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
 using System.Globalization;
+using System.Management.Automation;
 using System.Management.Automation.Host;
+using System.Security;
+using System.Text;
 using System.Threading;
 
-public class PuppetPSHost : PSHost
+namespace Puppet
 {
+  public class PowerShellRawUserInterface : PSHostRawUserInterface
```

Comment:
Minor - why add the namespace?  Why not just keep the class name consistent to avoid collisions (or is there some concern over polluting namespace)?

* `PuppetPSHost` exists derived from `PSHost` ... would think then that we use:
* `PuppetPSHostRawUserInterface` derived from `PSHostRawUserInterface`